### ISSUE TITLE
1.12.3 sbt test runner does not run all tests having the same name

### ISSUE THREAD

**Issue by mallman** [rickynils/scalacheck#166]
Running test code

    import org.scalacheck._

    object StringSpecification extends Properties("String") {
      import Prop.forAll
      property("substring") = forAll { (a: String, b: String) =>
        (a+b).substring(a.length) == b
      }

      property("substring") = forAll { (a: String, b: String, c: String) =>
        (a+b+c).substring(a.length, a.length+b.length) == b
      }
    }

in sbt 0.13.8 with ScalaCheck 1.12.2 prints

    > core/test
    [info] + String.substring: OK, passed 100 tests.
    [info] + String.substring: OK, passed 100 tests.
    [info] Passed: Total 2, Failed 0, Errors 0, Passed 2
    [success] Total time: 0 s, completed May 22, 2015 4:01:01 PM

but with 1.12.3 prints

    > core/test
    [info] + String.substring: OK, passed 100 tests.
    [info] Passed: Total 1, Failed 0, Errors 0, Passed 1
    [success] Total time: 0 s, completed May 22, 2015 4:02:10 PM

**Comment by rickynils** [comment]
Yeah, that is confusing. Do you feel there's a need for supporting this use case, or should ScalaCheck simply refuse adding properties with the same name to a `Properties` collection?

**Comment by mallman** [comment]
My main concern is that 12.2.3 skips tests that ran in 12.2.2. While it can be spotted when you have 2 or 3 tests in your codebase or are developing new ones, if you have a large test base this regression can easily go undetected.

Based on the example in the documentation and the behavior I found in 12.2.2, I assumed this behavior was supported. Whether or not that assumption is correct, I'm probably not the only one who made it.

In my particular situation, I'm using this feature to define multiple independent tests for a single property. For example, I've defined an ordering on a type and I have a few tests that look like this:

    property("<") = "does this" |: forAll...
    property("<") = "does that" |: forAll...

My reasoning in structuring my tests this way being that I'm testing different aspects of a single property.

For these reasons I suggest reinstating the previous behavior in subsequent releases of ScalaCheck.

Thanks for your time, and thanks for ScalaCheck. Cheers.
